featurizers/fingerprints.py

```diff
@@ -336,46 +339,48 @@ def __init__(
         self.pool_multiplier = pool_multiplier
         self.rmsd_cutoff = rmsd_cutoff
         self.max_energy_diff = max_energy_diff
-        self.force_field = force_field
+        self.forcefield = forcefield
         self.get_values = get_values
         self.is_folded = is_folded
         self.fold_bits = fold_bits
         self.standardise = standardise
-        self.random_state = random_state
+        self.seed = seed
 
     def _calculate_fingerprint(
         self, X: Union[pd.DataFrame, np.ndarray]
-    ) -> np.ndarray:
+    ) -> Union[np.ndarray, spsparse.csr_array]:
         from e3fp.conformer.generator import ConformerGenerator
-        from e3fp.conformer.util import mol_from_smiles
         from e3fp.pipeline import fprints_from_mol
 
-        X = self._validate_input(X)
-
         conf_gen = ConformerGenerator(
             first=self.first,
             num_conf=self.num_conf,
             pool_multiplier=self.pool_multiplier,
             rmsd_cutoff=self.rmsd_cutoff,
             max_energy_diff=self.max_energy_diff,
-            forcefield=self.force_field,
+            forcefield=self.forcefield,
             get_values=self.get_values,
-            seed=self.random_state,
+            seed=self.seed,
         )
 
         result = []
 
         for x in X:
-            input_mol = mol_from_smiles(
-                smiles=MolToSmiles(x),
-                name=MolToSmiles(x),
-                standardise=self.standardise,
-            )
+            if isinstance(x, Mol):
+                smiles = MolToSmiles(x)
+                mol = x
+            else:
+                smiles = x
+                mol = MolFromSmiles(x)
+
+            mol.SetProp("_Name", smiles)
+            if self.standardise:
+                mol = mol_to_standardised_mol(mol)
```

Comment:
Standardization of molecules is a very typical preprocessing step and while we should probably make a utility function (as a transformer class) of it, it shouldn't be managed by this class. Standardized molecules can be used for any fingerprints, after all.

Reply:
Removed. 

We could implement a standarisation helper function in the base class that would work similarily to _validate_input and that could also allow to standarise molecules concurrently. Also TODO as future work?